### PR TITLE
[Testing:Developer] Manual parallelizing in Github Actions

### PR DESCRIPTION
**Pull request by IDzyre**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
Currently, Cypress Cloud parallelizes the runs, however this only works for Submitty members, not external contributors. It also has a limit of 100,000 different test runs (Specific `It('should test...', ()=>{});` runs, not spec files or describe instances). With increasing development, and the creation of new spec files that add to the tests, this will most likely not be enought. 

### What is the new behavior?
The runs are now split up into _five_ different categories manually, and Cypress Cloud has been removed from CI:

- Admin -- Testing admin features, or website options
- Feature -- Testing user features, polls, forums, late day caching, etc.
- Gradeable -- Testing creation, editing, submission of gradeables
- System -- Testing things that need other system functions to run, and accessibility testing. 
- UI -- Testing misc user interfaces

### CODE REVIEW

**Review by jason490 — CHANGES_REQUESTED** [review]
You can try using the github matrix to get rid of doing similar operations
ex:
```
Cypress-Test:
    ...
   strategy:
      matrix:
           containers: ["gradeable","admin","UI",etc...]
   ...
   ...
         spec: cypress/e2e/${{matrix.containers}}
```

**Review by jason490 — APPROVED** [review]

File: .github/workflows/submitty_ci.yml [hunk 1]
```diff
@@ -468,11 +474,9 @@ jobs:
         with:
           config: baseUrl=http://localhost
           working-directory: ${{env.SUBMITTY_REPOSITORY}}/site
-          spec: cypress/e2e/cypress-feature/*
+          spec: cypress/e2e/Cypress-${{matrix.containers}}/*
           browser: chrome
           env: run_area=CI
-          record: true
-          parallel: true  # Runs test in parallel using settings above
```
Comment (williamjallen) [inline .github/workflows/submitty_ci.yml:475]: (Leaving this here since I can't comment on lines you didn't edit)

Are the env variables under this necessary now?

**Review by williamjallen — COMMENTED** [review]
Changes seem reasonable to me.  I do worry about needing to rebalance the sets of tests occasionally, but that's certainly manageable, and better than the alternative of just running all of the tests sequentially.

Is there any parallelization you can do in one or more of the test sets?  We'd have to make sure none of the tests potentially interfere with one another and the benefits of parallelization would be hampered by the limited resources available on the CI runners, but it may reduce the time slightly at least.

File: .github/workflows/submitty_ci.yml (hunk 1 above)
Comment (IDzyre) [reply .github/workflows/submitty_ci.yml:475]: No they are not, I will remove them tonight.

**Comment by IDzyre** [comment]
> Changes seem reasonable to me. I do worry about needing to rebalance the sets of tests occasionally, but that's certainly manageable, and better than the alternative of just running all of the tests sequentially.
> 
> Is there any parallelization you can do in one or more of the test sets? We'd have to make sure none of the tests potentially interfere with one another and the benefits of parallelization would be hampered by the limited resources available on the CI runners, but it may reduce the time slightly at least.

I think its probably possible, but to make the CI run more smoothly as it is, I think that can be a future PR.

**Review by williamjallen — APPROVED** [review]
Good work on this!

**Review by jason490 — APPROVED** [review]
